### ISSUE TITLE
Modular Rules System

### ISSUE THREAD

**Issue by ruler501** [magefree/mage#4705]
While reading issue #4696 I started thinking that a modular rules/variants system that lets you add and remove modifications like commander in command zone, or color identity, singleton, etc could be very useful.

It would allow players to enforce deck building and rules restrictions for the most part instead of just going to singleton. This could make playing more casual unofficial formats easier.

Things that I've seen as potential rules that could be added and removed are:
Commander in command zone(ensuring there is only one card in sideboard)
Color identity of cards in deck subset of the commanders(requires only one card in sideboard)
Allowed commander card types(legendary creature default, could also be planeswalker or arbitrary legendary or arbitrary creature)
Card rarity restrictions(Common only for pauper like formats for instance)
Minimum deck size
Maximum deck size
Max Copies of a card in a deck
Starting life total
Starting hand size
Poison counter kill amount
Custom ban list

**Comment by Zzooouhh** [comment]
Modular rules for freeform formats could be a good idea, however it should stay in freeform only.

**Comment by ruler501** [comment]
I fully agree and it should be completely obvious to anyone trying to join what exactly the rules changes are to reduce the chances of someone modifying it to be unfair without informing the other players. It could be its own custom format name in the games listing, something like Freeform would work if it wasn't already used or even just Modular.
